afs/afs-storage-api/src/main/java/com/powsybl/afs/storage/InMemoryEventsBus.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.afs.storage;
+
+import com.powsybl.afs.storage.events.AppStorageListener;
+import com.powsybl.afs.storage.events.NodeEvent;
+import com.powsybl.afs.storage.events.NodeEventList;
+import com.powsybl.commons.util.WeakListenerList;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * @author Chamseddine Benhamed <chamseddine.benhamed at rte-france.com>
+ */
+public class InMemoryEventsBus implements EventsBus {
```

Comment:
Would it be possible to have some kind of autoflush ?
(there is a flush potentially missing in the `NodeEventServer` class)
Like trigger a delayed flush after pushEvent if flush hasn't been called in the meantime.

Reply:
We don't have to add flush method in the NodeEventServer :  storage.flush() in RemoteAppStorage calls StorageChangeBuffer's flush method which calls the flush method in the remote app storage, then the execution of that method sends the events list from the NodeEvent Server (as it's a listener of the remote app storage event bus) to the NodeEventClient  which notify all local listeners.

Autoflush -> flush after each push event , is it really beneficial ?
here we should decide, batch versus continuous treatment, it depends on uses cases.
@geofjamg  what do you think ?